packages/plugins/src/plugins/glint-fix.plugin.ts

```diff
@@ -135,15 +168,21 @@ export class GlintFixPlugin extends Plugin<GlintFixPluginOptions> {
           if (change.span.length > 0) {
             changeTracker.remove(change.span.start, change.span.start + change.span.length);
           }
+          const targetFileName = fileTextChange.fileName;
 
-          if (!this.appliedAtOffset[fileTextChange.fileName]) {
-            this.appliedAtOffset[fileTextChange.fileName] = [];
-          }
-
-          if (this.appliedAtOffset[fileTextChange.fileName].includes(change.span.start)) {
-            continue;
+          // Track the applied changes so we can dedupe as we go
+          if (this.hasAppliedChange(targetFileName, change)) {
+            continue; // Skip if a duplicate change
           } else {
-            // this.appliedAtOffset[fileTextChange.fileName].push(change.span.start);
+            // Init
+            if (!this.appliedTextChanges[targetFileName]) {
+              this.appliedTextChanges[targetFileName] = [];
```

Comment:
We have something new [for this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment):

```
this.appliedTextChanges[targetFileName] ??= []
```